dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointMetaInformationManager.kt

```diff
@@ -0,0 +1,95 @@
+    fun getDataPointMetaInformationByDataId(dataId: String): DataPointMetaInformationEntity =
+        dataPointMetaInformationRepositoryInterface.findById(dataId).orElseThrow {
+            ResourceNotFoundApiException(
+                "Data point not found",
+                "No data point with the id: $dataId could be found in the data store.",
+            )
+        }
+
+    /**
+     * Method to get the currently active data id for a specific data point dimension
+     * @param dataPointDimensions the data point dimension to get the currently active data id for
+     * @return the id of the currently active data point
+     */
+    fun getCurrentlyActiveDataId(dataPointDimensions: DataPointDimensions): String? =
+        dataPointMetaInformationRepositoryInterface.getActiveDataPointId(dataPointDimensions)
+
+    /**
+     * Method to get the data point dimension from a data id
+     * @param dataId the id of the data point
+     * @return the data point dimension
+     */
+    fun getDataPointDimensionFromId(dataId: String): DataPointDimensions {
+        val dataPointMetaInformation = getDataPointMetaInformationByDataId(dataId)
+        return DataPointDimensions(
+            reportingPeriod = dataPointMetaInformation.reportingPeriod,
+            companyId = dataPointMetaInformation.companyId,
+            dataPointIdentifier = dataPointMetaInformation.dataPointIdentifier,
+        )
+    }
+
+    /**
+     * Method to store the meta information of a data point
+     * @param dataPointMetaInformation the meta information to store
+     * @return the stored meta information
+     */
+    @Transactional
+    fun storeDataPointMetaInformation(dataPointMetaInformation: DataPointMetaInformationEntity): DataPointMetaInformationEntity =
+        dataPointMetaInformationRepositoryInterface.save(dataPointMetaInformation)
+
+    /**
+     * Method to update the QA status of a data point
+     * @param dataId the id of the data point to update
+     * @param newQaStatus the new value for the QA status
+     */
+    @Transactional
+    fun updateQaStatusOfDataPoint(
+        dataId: String,
+        newQaStatus: QaStatus,
+    ) {
+        val dataPointMetaInformation = getDataPointMetaInformationByDataId(dataId)
+        dataPointMetaInformation.qaStatus = newQaStatus
+        dataPointMetaInformationRepositoryInterface.save(dataPointMetaInformation)
+    }
+
+    /**
+     * Method to update the currently active flag of a data point
+     * @param dataId the id of the data point to update
+     * @param newCurrentlyActiveValue the new value for the currently active flag
+     */
+    @Transactional
+    fun updateCurrentlyActiveFlagOfDataPoint(
+        dataId: String?,
+        newCurrentlyActiveValue: Boolean,
+    ) {
+        if (dataId == null) {
+            return
+        }
+        val dataPointMetaInformation = getDataPointMetaInformationByDataId(dataId)
+        dataPointMetaInformation.currentlyActive = newCurrentlyActiveValue
```

Comment:
Das klappt nicht. Es darf niemals "false" in der Datenbank für currentlyActive stehen, sondern immer entweder "true" oder "null"

Reply:
Das sollten wir mal besprechen. Ist mir nicht ganz klar wo das Problem liegen soll.

Reply:
fixed